core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/RawInfraImpl.kt

```diff
@@ -106,33 +115,66 @@ class RawInfraImpl(
     val logicalSignalPool: StaticPool<LogicalSignal, LogicalSignalDescriptor>,
     val physicalSignalPool: StaticPool<PhysicalSignal, PhysicalSignalDescriptor>,
     val zonePathPool: StaticPool<ZonePath, ZonePathDescriptor>,
-    val zonePathMap: Map<ZonePathSpec, ZonePathId>
+    val zonePathMap: Map<ZonePathSpec, ZonePathId>,
+    val operationalPointPartPool: StaticPool<OperationalPointPart, OperationalPointPartDescriptor>,
+    val trackSectionIdMap: Map<String, TrackSectionId>,
 ) : RawInfra {
     override val trackNodes: StaticIdxSpace<TrackNode>
         get() = trackNodePool.space()
     override val trackSections: StaticIdxSpace<TrackSection>
         get() = trackSectionPool.space()
 
-    override fun getTrackSectionDetectors(trackSection: TrackSectionId): StaticIdxList<Detector> {
-        return trackSectionPool[trackSection].detectors
-    }
-
     override fun getTrackSectionId(trackSection: TrackSectionId): String {
         return trackSectionPool[trackSection].name
     }
 
+    override fun getTrackSectionFromId(id: String): TrackSectionId? {
+        return trackSectionIdMap[id]
+    }
```

Comment:
so far, string identifiers were called "name" in the API, to disambiguate with numerical IDs